extensions/amp-story/1.0/story-analytics.js

```diff
@@ -76,26 +76,34 @@ export class StoryAnalyticsService {
 
     /** @const @private {!./variable-service.AmpStoryVariableService} */
     this.variableService_ = getVariableService(win);
+
+    /** @private @const {!./amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = getStoreService(win);
+
+    this.initializeListeners_();
   }
 
-  /**
-   * @param {!./navigation-state.StateChangeEventDef} stateChangeEvent
-   */
-  onNavigationStateChange(stateChangeEvent) {
-    switch (stateChangeEvent.type) {
-      case StateChangeType.ACTIVE_PAGE:
-        this.triggerEvent(AnalyticsEvent.PAGE_VISIBLE);
-        break;
-      case StateChangeType.BOOKEND_ENTER:
-        this.triggerEvent(AnalyticsEvent.BOOKEND_ENTER);
-        break;
-      case StateChangeType.BOOKEND_EXIT:
-        this.triggerEvent(AnalyticsEvent.BOOKEND_EXIT);
-        break;
-      case StateChangeType.LAST_PAGE:
+  /** @private */
+  initializeListeners_() {
+    this.storeService_.subscribe(StateProperty.BOOKEND_STATE, isActive => {
+      this.triggerEvent(
+        isActive ? AnalyticsEvent.BOOKEND_ENTER : AnalyticsEvent.BOOKEND_EXIT
+      );
+    });
+
+    // Using pageId instead of pageIndex since it's guaranteed to be called at
```

Comment:
+1 to using pageId, but for different reasons: since pages can be appended by a live list, indexes wouldn't be reliable.
For the "it's guaranteed to be called", we should use that extra parameter `callToInitialize`.

Reply:
Done.